From a00df6b03955262e619073eeb84eb8f2bdc1d4e4 Mon Sep 17 00:00:00 2001
From: Dimitry Ishenko <dimitry.ishenko@gmail.com>
Date: Mon, 2 Oct 2023 14:58:25 -0400
Subject: [sdas] Increase address length to 24 bits

---
 sdas/asrab/rabmch.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sdas/asrab/rabmch.c b/sdas/asrab/rabmch.c
index 53450f0..37c0148 100644
--- a/sdas/asrab/rabmch.c
+++ b/sdas/asrab/rabmch.c
@@ -1347,6 +1347,11 @@ minit(void)
          */
         hilo = 0;
 
+        /*
+         * Address Space
+         */
+        exprmasks(3);
+
         if (pass == 0) {
                 mchtyp = X_R2K;
                 sym[2].s_addr = X_R2K;
-- 
2.39.2

