From a187caad5059c6c54686aedd9e1f1382cba2f79c Mon Sep 17 00:00:00 2001
From: Dimitry Ishenko <dimitry.ishenko@gmail.com>
Date: Tue, 3 Oct 2023 17:02:03 -0400
Subject: [rab] Fix ISR return instruction

---
 src/z80/gen.c | 16 +++++-----------
 1 file changed, 5 insertions(+), 11 deletions(-)

diff --git a/src/z80/gen.c b/src/z80/gen.c
index 1e42cfa..da308b7 100644
--- a/src/z80/gen.c
+++ b/src/z80/gen.c
@@ -6656,10 +6656,8 @@ genEndFunction (iCode *ic)
          If critical function then turn interrupts back on */
       if (IFFUNC_ISCRITICAL (sym->type))
         {
-          if (IS_SM83 || IS_TLCS90)
+          if (IS_SM83 || IS_TLCS90 || IS_RAB)
             emit2 ("!ei");
-          else if (IS_RAB)
-            emit2 ("ipres");
           else
             {
               symbol *tlbl = newiTempLabel (NULL);
@@ -6797,9 +6795,10 @@ genEndFunction (iCode *ic)
     {
       if (is_nmi)
         emit2 ("retn");
-      else if (IS_RAB && IFFUNC_ISCRITICAL (sym->type) && FUNC_INTNO (sym->type) == INTNO_UNSPEC)
+      else if (IS_RAB)
         {
-          emit2 ("ipres");
+          if (IFFUNC_ISCRITICAL (sym->type))
+            emit2 ("!ei");
           emit2 ("ret");
         }
       else if (IS_SM83)
@@ -15113,16 +15112,11 @@ genEndCritical (const iCode * ic)
 {
   symbol *tlbl = regalloc_dry_run ? 0 : newiTempLabel (0);
 
-  if (IS_SM83 || IS_TLCS90)
+  if (IS_SM83 || IS_TLCS90 || IS_RAB)
     {
       emit2 ("!ei");
       regalloc_dry_run_cost += 1;
     }
-  else if (IS_RAB)
-    {
-      emit2 ("ipres");
-      regalloc_dry_run_cost += 1;
-    }
   else if (IC_RIGHT (ic))
     {
       aopOp (IC_RIGHT (ic), ic, FALSE, TRUE);
-- 
2.39.2

